privacy/enforcement.go

```diff
@@ -26,17 +25,33 @@ func (e Enforcement) Apply(bidRequest *openrtb.BidRequest, ampGDPRException bool
 
 func (e Enforcement) apply(bidRequest *openrtb.BidRequest, ampGDPRException bool, scrubber Scrubber) {
 	if bidRequest != nil && e.Any() {
-		bidRequest.Device = scrubber.ScrubDevice(bidRequest.Device, e.getIPv6ScrubStrategy(), e.getGeoScrubStrategy())
+		bidRequest.Device = scrubber.ScrubDevice(bidRequest.Device, e.getDeviceIDScrubStrategy(), e.getIPv4ScrubStrategy(), e.getIPv6ScrubStrategy(), e.getGeoScrubStrategy())
 		bidRequest.User = scrubber.ScrubUser(bidRequest.User, e.getUserScrubStrategy(ampGDPRException), e.getGeoScrubStrategy())
 	}
 }
 
+func (e Enforcement) getDeviceIDScrubStrategy() ScrubStrategyDeviceID {
+	if e.COPPA || e.GDPRID || e.CCPA || e.LMT {
+		return ScrubStrategyDeviceIDAll
+	}
+
+	return ScrubStrategyDeviceIDNone
+}
+
+func (e Enforcement) getIPv4ScrubStrategy() ScrubStrategyIPV4 {
+	if e.COPPA || e.GDPRGeo || e.CCPA || e.LMT {
+		return ScrubStrategyIPV4Lowest8
+	}
+
+	return ScrubStrategyIPV4None
```

Comment:
This was always set to `ScrubStrategyIPV4Lowest8 ` before.  Now it resolves to None for GDPR if only ID is being enforced. IP rounding is considered Geo enforcement.